### PR TITLE
[DPP-1363] Handle outstanding ETQ TODOs (part 1): Cleanup ACS reader 

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    maxPageSize: Int,
+) {
+  assert(minPageSize > 0)
+  assert(maxPageSize >= minPageSize)
+}
+
+object IdPageSizing {
+  private val logger = ContextualizedLogger.get(getClass)
+
+  def calculateFrom(
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala:21]: This method was just moved without changes

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/FilterUtils.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao.events
+
+import com.daml.platform.{Identifier, Party, TemplatePartiesFilter}
+
+case class DecomposedFilter(party: Party, templateId: Option[Identifier])
+
+object FilterUtils {
+  def decomposeFilters(filter: TemplatePartiesFilter): Seq[DecomposedFilter] = {
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/FilterUtils.scala:11]: This method was just moved without changes

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventIdsUtils.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    else
+      mergeSort(
+        sources
+          .drop(2)
+          .appended(
+            sources.take(2).reduce(_ mergeSorted _)
+          )
+      )
+
+  protected[events] def statefulDeduplicate[T]: () => T => List[T] =
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventIdsUtils.scala:31]: This method was just moved without changes

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventIdsUtils.scala [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+import scala.annotation.tailrec
+
+object EventIdsUtils {
+
+  def sortAndDeduplicateIds(ids: Vector[Source[Long, NotUsed]]): Source[Long, NotUsed] =
+    mergeSort(ids).statefulMapConcat(statefulDeduplicate)
+
+  @tailrec
+  protected[events] def mergeSort[T: Ordering](
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventIdsUtils.scala:17]: This method was just moved without changes

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/PaginatingAsyncStream.scala [hunk 5]
```diff
@@ -79,35 +81,34 @@ private[platform] object PaginatingAsyncStream {
       .flatMapConcat(Source(_))
   }
 
-  case class IdPaginationState(startOffset: Long, pageSize: Int)
+  case class IdPaginationState(fromIdExclusive: Long, pageSize: Int)
 
-  // TODO etq: Unify with com.daml.platform.store.dao.events.FilterTableACSReader.idSource
   def streamIdsFromSeekPagination(
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/PaginatingAsyncStream.scala:86]: This is now used also by the ACS reader

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ACSReader.scala [hunk 6]
```diff
@@ -122,140 +116,26 @@ class FilterTableACSReader(
...
-      } else {
-        logger.debug(
-          s"Calculated maximum ID page size supporting API stream memory limits [$calculatedMaxIdPageSize] is sufficiently high, using [$maxIdPageSize] instead."
-        )
-        IdQueryConfiguration(RecommendedMinIdPageSize, maxIdPageSize)
-      }
-    }
-  }
-
-  def idSource(
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/ACSReader.scala:214]: This is deleted and the equivalent utility from PaginatngAsyncStream is used.

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/PaginatingAsyncStream.scala [hunk 7]
```diff
@@ -79,35 +81,34 @@ private[platform] object PaginatingAsyncStream {
...
           case empty if empty.isEmpty => None
-          case nonEmpty: Vector[Long] =>
-            val nextPageStartOffset = nonEmpty.last
-            val newState = IdPaginationState(
-              startOffset = nextPageStartOffset,
-              pageSize = Math.min(state.pageSize * 4, pageConfig.maxPageSize),
+          case ids =>
+            val nextState = IdPaginationState(
+              fromIdExclusive = ids.last,
+              pageSize = Math.min(state.pageSize * 4, idPageSizing.maxPageSize),
```
Comment (skisel-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/PaginatingAsyncStream.scala:106]: I am wondering, what is 4?

**Review by skisel-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/PaginatingAsyncStream.scala (hunk 7 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/PaginatingAsyncStream.scala:106]: We fetch event ids in pages starting with a very small page size for low latency and then transition into larger page sizes for throughput.
4 is just a multiplier of the page size we use to do that.

**Pull request merged**